lib/charms/tls_certificates_interface/v4/tls_certificates.py

```diff
@@ -989,6 +1013,10 @@ def __init__(
             mode (Mode): Whether to use unit or app certificates mode. Default is Mode.UNIT.
             refresh_events (List[BoundEvent]): A list of events to trigger a refresh of
               the certificates.
+            private_key (Optional[PrivateKey]): The private key to use for the certificates.
+                If provided, it will be used instead of generating a new one.
+                If the key is not valid the library will generate a new one.
```

Comment:
Is that true? If the key is not valid, we raise (which is the right thing to do in my opinion) and we do not generate a new one

Reply:
Updated.